src/service/viewport-impl.js

```diff
@@ -584,15 +584,15 @@ export class Viewport {
   }
 
   /**
-   * @param {!JSONType} event
+   * @param {!JSONType} data
    * @private
    */
-  updateOnViewportEvent_(event) {
-    const paddingTop = event['paddingTop'];
-    const duration = event['duration'] || 0;
-    const curve = event['curve'];
+  updateOnViewportEvent_(data) {
+    const paddingTop = data['paddingTop'];
```

Comment:
According to old code, it's possible that `paddingTop` is `undefined.` I.e. see this old code snippet:
```
    if (eventType == 'viewport') {
      if (data['paddingTop'] !== undefined) {
        this.paddingTop_ = data['paddingTop'];
        this.viewportObservable_.fire(
          /** @type {!JSONType} */ (data));
```
I don't know if it's a real situation here. If it is, we may want to add the following in this method:
```
const paddingTop = data['paddingTop'];
if (paddingTop == undefined) {
  return;
}
```
Please check.